mlir/include/mlir/Dialect/LLVMIR/GENXOps.td

```diff
@@ -326,4 +322,80 @@ def GENX_AtomicRMWOp : GENX_Op<"atomic.rmw", [
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// Joint Matrix operations
+//===----------------------------------------------------------------------===//
+
+def GENX_IsJointMatrixType :
+  CPred<"$_self.isa<::mlir::GENX::JointMatrixType>()">;
+
+def GENX_AnyJointMatrix : DialectType<GENX_Dialect, GENX_IsJointMatrixType,
+                                "any GENX joint matrix type">;
+
+def ScopeKind : I32EnumAttr<"Scope", "GENX Scope", 
```

Comment:
Does it make sense to merge with `MemoryScope`?

Reply:
I think it does, we should have one attribute in the GENX dialect for Scope. 
I think is better to do this in a separate PR. We have to prob. also change the codegen in `createFence` to call a SPIRV function that the translator will change to a SPIRV `OpMemoryBarrier`

Reply:
ok